lib/devise_two_factor/models/two_factor_authenticatable.rb

```diff
@@ -32,7 +32,9 @@ def validate_and_consume_otp!(code, options = {})
         return false unless code.present? && otp_secret.present?
 
         totp = self.otp(otp_secret)
-        return consume_otp! if totp.verify_with_drift(code, self.class.otp_allowed_drift)
+        if totp.verify(code, drift_ahead: self.class.otp_allowed_drift, drift_behind: self.class.otp_allowed_drift)
```

Comment:
You can see [here](https://github.com/mdp/rotp/commit/15dab263022e51b2c0646df4cb2f5d6cff99cdd4#diff-ed386b7f6dadc0ef09bf81c543cffbfbL42) that they just split this `drift` arg into ahead/behind.